HW-5 MVVM Review 적용

BeokBeok opened the pull request:
MVVM 적용 하였습니다.

BeokBeok committed:
HW-4 DataBinding Gradle 설정

BeokBeok committed:
Revert "HW-4 DataBinding Gradle 설정"

This reverts commit 099cb5a36d76828f7367a1b0b8573a1abad0381e.

BeokBeok committed:
HW-4 ConstraintLayout Depth 정리

Commit:
Merge remote-tracking branch 'origin/class02/BeokBeok' into class02/BeokBeok

BeokBeok committed:
HW-4 DataBinding gradle 설정

BeokBeok committed:
HW-4 코틀린 버전 업데이트

BeokBeok committed:
HW-4 MVVM 패턴 적용

BeokBeok committed:
HW-4 View Model 에서 View 를 접근하지 않도록 변경 및 Divider 제거

BeokBeok committed:
HW-4 Err 발생 시 Toast 발생 기능 적용

- Throwable 로 처리하지 않는 경우,
메시지가 변경되지 않으면 Toast 발생하지 않음

BeokBeok committed:
HW-4 마켓 항목 클릭 시, 클릭된 View 만 색 변경 기능 적용

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitFragment.kt:
```diff
@@ -40,30 +34,34 @@ class UpbitFragment : BaseFragment(
-            )
+    private fun setBinding() {
+        val vm = UpbitViewModel(UpbitRepository(UpbitRetrofitDataSource(UpbitRetrofit.retrofit)))
+        binding.vm = vm
```
binding.vm = vm 의 역할이 binding.setVariable(BR.vm, vm) 이거랑 같은 건가요?

BeokBeok replied:
네, 맞습니다.

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitFragment.kt:
```diff
@@ -77,7 +75,6 @@ class UpbitFragment : BaseFragment(
 
         if (view is TextView) {
             view.setTextColor(Color.BLUE)
```
이곳에서 한 번에 컬러와 리스트를 처리 안 하신 것은 분리하기 위해서인가요?

BeokBeok replied:
filter와 map을 사용하여 선택된 뷰가 아닌 것들에 대하여 gray 처리 후, 선택된 뷰만 blue 처리하도록 하였습니다.

NackUn reviewed (dismissed):
고생하셨습니다!

NackUn reviewed (commented):
제가 Approve해도 되나요?

BeokBeok commented:
@NackUn 
석준님이 approve 하는 것으로 변경되었습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitViewModel.kt:
```diff
@@ -0,0 +1,38 @@
+
+    var marketPriceList = ObservableField<List<Map<String, String>>>()
+    var errMsg = ObservableField<Throwable>()
+    var selectedTextView = ObservableField<TextView>()
```
나중에 테스트나 이런것 때문에 ViewModel쪽에는 DataBinding을 위한 것을 외에는 TextView나 View와 같은 android 관련 패키지 내용이 들어오게 되면 안됩니다.

BeokBeok replied:
수정 하였습니다.
30abda1

해당 수정사항이 석준님이 의도한 수정사항인지는 잘 모르겠습니다.
만약, 의도하신 것이 아니라면 방향성만 제시 부탁 드립니다.

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitFragment.kt:
```diff
@@ -40,30 +34,34 @@ class UpbitFragment : BaseFragment(
+            object : Observable.OnPropertyChangedCallback() {
+                override fun onPropertyChanged(sender: Observable?, propertyId: Int) {
+                    if (sender is ObservableField<*>) {
+                        setViewColor((sender.get() as TextView))
```
요기 구분하고자 가로 2개 하신건가요??

BeokBeok replied:
텍스트뷰를 받지 않는 것으로 변경하면서 수정되었습니다.
30abda1

BeokBeok committed:
HW-5 불필요한 파일 제거

BeokBeok committed:
HW-5 TODO 주석 추가

BeokBeok committed:
HW-5 Kotlin 버전 업데이트

BeokBeok committed:
HW-5 클래스 구현 선언 관련 Line Break 적용

BeokBeok committed:
HW-5 BaseViewModel 추가

BeokBeok committed:
HW-5 onViewCreated -> onActivityCreated 로 변경

BeokBeok committed:
HW-5 BaseViewModel 관련 TODO 주석 제거

BeokBeok committed:
HW-5 패키지명 변경

- recyclerView -> adapter
- ext 패키지 추가

BeokBeok committed:
HW-5 선택된 TextView 만 색상 변경하는 부분 수정

- TextView 에서 Boolean 으로 변경
- setViewColor 함수 제거

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/92#discussion_r300736678

BeokBeok committed:
HW-5 GetTickerCallback 제거

BeokBeok committed:
HW-5 접근자 변경 (var -> val)

BeokBeok committed:
HW-5 enum 코드 정리

BeokBeok committed:
HW-5 Test 라이브러리 추가

BeokBeok committed:
HW-5 접근자 추가 변경 (var -> val)

BeokBeok committed:
HW-5 itemView 데이터바인딩 1차 적용

BeokBeok committed:
HW-5 TODO 제거 및 gradle 정리

BeokBeok committed:
HW-5 Progress Bar 추가

BeokBeok committed:
HW-5 kotlin-android-extensions 제거

NackUn reviewed (commented):
callback 제거를 higher order function 이 아닌 observe를 이용해서 하셨군요

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/data/upbit/UpbitRepository.kt:
```diff
@@ -1,29 +1,18 @@
-        callback: UpbitDataSource.GetTickerCallback
-    ) {
+    val marketPriceList = ObservableField<List<UpbitTicker>>()
+    val throwable = ObservableField<Throwable>()
```
repository에 OservableField가 있으면 안됩니다.
저번에 말했던것처럼 higher-order function으로 데이터 전달하게 해주세요

BeokBeok replied:
c72640b 
적용 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/data/upbit/source/UpbitRetrofitDataSource.kt:
```diff
@@ -1,19 +1,18 @@
-        prefix: String,
-        callback: UpbitDataSource.GetTickerCallback
-    ) {
+    private val retrofit = UpbitRetrofit.retrofit
```
retrofit을 생성자쪽에서 받게 해주세요

BeokBeok replied:
773eb43 
적용 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/data/upbit/source/UpbitRetrofitDataSource.kt:
```diff
@@ -1,19 +1,18 @@
-        callback: UpbitDataSource.GetTickerCallback
-    ) {
+    private val retrofit = UpbitRetrofit.retrofit
+    private val upbitRepository = UpbitRepository(this)
```
DataSource는 Repository를 알면 안됩니다.
이 코드는 제거해주세요.

BeokBeok replied:
c72640b 
적용 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitFragment.kt:
```diff
@@ -48,33 +45,22 @@ class UpbitFragment : BaseFragment<FragmentUpbitBinding>(
-                    if (sender is ObservableField<*>) {
-                        setViewColor((sender.get() as TextView))
+                    if (sender is ObservableField<*> && sender.get() == false) {
+                        binding.tvMarketKrw.setTextColor(Color.GRAY)
```
color을 set해주는 부분은 xml에서 처리해주세요

BeokBeok replied:
ae840a6 
적용 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitViewModel.kt:
```diff
@@ -1,38 +1,62 @@
-            upBitRepository.getMarketPrice(v.text.toString(), this)
+            isLoading.set(true)
+            upBitRepository.getMarketPrice(v.text.toString())
+            v.setTextColor(Color.BLUE)
```
View나 TextView가 ViewModel에 있으면 안됩니다.
해당부분은 xml에서 처리해주세요

BeokBeok replied:
ae840a6 
적용 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/adapter/UpbitAdapter.kt:
```diff
@@ -32,16 +33,13 @@ class UpbitAdapter : RecyclerView.Adapter<UpbitAdapter.ViewHolder>() {
-            currentPrice.text = marketPrice["currentPrice"]
-            netChange.text = marketPrice["netChange"]
-            tradingVal.text = marketPrice["tradingVal"]
+            binding.tvCoinName.text = marketPrice["coinName"]
```
text을 set해주는 부분은 xml에 있으면 됩니다.

BeokBeok replied:
df63aa6
적용 하였습니다.

BeokBeok committed:
HW-5 Callback 처리를 ObservableField 가 아닌, Higher-Order-Function 처리

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/92#discussion_r302166063
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/92#discussion_r302166608

BeokBeok committed:
HW-5 retrofit 을 생성자에서 받도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/92#discussion_r302166474

BeokBeok committed:
HW-5 Text Color 를 xml 에서 설정 및 View Model 에서 View 가 없도록 수정

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/92/files/85b7e3a5b9bf01e5447ca065c7d4765b8badb946#r302166858
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/92/files/85b7e3a5b9bf01e5447ca065c7d4765b8badb946#r302167070

BeokBeok committed:
HW-5 RecyclerViewExt 파일명 및 패키지 위치 변경

BeokBeok committed:
HW-5 item bind 시 xml 에서 set 하도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/92/files/85b7e3a5b9bf01e5447ca065c7d4765b8badb946#r302167360

BeokBeok committed:
HW-5 unused Annotation 추가

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋
